packages/babel-generator/src/printer.js

```diff
@@ -58,7 +58,7 @@ export default class Printer extends Buffer {
     if (needsParens) this.push(")");
 
     // end
-    this.map.mark(node, "end");
+    if (parent) this.map.mark(parent);
```

Comment:
@kpdecker This change has introduced an issue and I'm trying to verify, was this changed to fix a specific issue that you were seeing when debugging? I agree that the map output before this also wasn't the best, but this seems worse.

I've got some fixes locally, so I'll tag you in any PRs I post, but I wanted your thoughts first.